corehq/apps/case_importer/tracking/filestorage.py

```diff
@@ -0,0 +1,94 @@
+import os
+from tempfile import mkstemp
+import uuid
+from django.core.cache import caches
+from corehq.apps.case_importer.tracking.models import CaseUploadFileMeta
+from corehq.blobs import get_blob_db
+from corehq.blobs.util import random_url_id
+from corehq.util.files import file_extention_from_filename
+from dimagi.utils.decorators.memoized import memoized
+
+BUCKET_PREFIX = 'case_importer'
+
+
+class PersistentFileStore(object):
+    def __init__(self, bucket, meta_model):
```

Comment:
As I think you could tell, I wrote this from the perspective of what general storage interface I wish we had, and I imagined that different apps that used it would want to store the meta-data in different tables.

> Yea, I agree with that, it would be great to separate those two use cases.
I think if you're proposing a this as a replacement for the file storage /
download expose part of soil then it should be in a separate app and not
part of case importer.

I can certainly do that, if you think this is good. If I do that, it'd make it more important for me to answer the question of whether different apps using this should have their meta data in different tables. The plus side is that makes it pretty easy to see things like the combined size of all case imports, that's maybe not that important and you can get that with a join assuming the tables aren't so huge that that becomes infeasible, which we probably don't have to worry about for this feature. Is it fine practice for lots of different models to contain foreign keys to the same central table? (So `CaseUploadRecord` would have a foreign key to `PersistentFileStorageMeta`, as would lots of other tables?)

If you don't think it's worth me generalizing this, then I can punt that question and make this more explicitly. If you do think it's worth it, maybe this can undergo architecture review.